Solar-System.rb

```diff
@@ -0,0 +1,225 @@
+    @name = name
+    @diameter = diameter
+    @gravity = gravity
+    @year_length = year_length
+    @distance_from_the_sun = distance_from_the_sun
+  end
+
+  def show_attributes
+    plant_attributes += "#{@name}'s attributes:"
+    plant_attributes += "\nIt has a diameter of #{@diameter}."
+    plant_attributes += "\nIts gravity is #{@gravity}."
+    plant_attributes += "\nIt takes #{@year_length} to spin on Axis."
+    plant_attributes += "\nIts distance from the sun is #{@distance_from_the_sun}.\n"
+    return plant_attributes
+  end
+
+end
+
+# The planet pool is an array holding hashes of planet info.
+pool = [
+  {
+    name: "Mercury",
+    diameter: "3,301 miles",
+    gravity: 0.36,
+    year_length: "59 days",
+    distance_from_the_sun: "36000000 miles",
+  },
+  {
+    name: "Venus",
+    diameter: "7,521 miles",
+    gravity: 0.9,
+    year_length: "243 days",
+    distance_from_the_sun: "67000000 miles",
+  },
+  {
+    name: "Earth",
+    diameter: "12,765 miles",
+    gravity: 1,
+    year_length: "23 hours, 56 mins",
+    distance_from_the_sun: "92960000 miles",
+  },
+  {
+    name: "Jupiter",
+    diameter: "88,846 miles",
+    gravity: 2.64,
+    year_length: "9 hours, 55 mins",
+    distance_from_the_sun: "483500000 miles",
+  },
+  {
+    name: "Saturn",
+    diameter: "74,900 miles",
+    gravity: 1.16,
+    year_length: "10 hours, 39 mins",
+    distance_from_the_sun: "888750000 miles",
+  },
+  {
+    name: "Uranus",
+    diameter: "31,763 miles",
+    gravity: 1.11,
+    year_length: "17 hours, 14 mins",
+    distance_from_the_sun: "1783744000 miles",
+  },
+  {
+    name: "Neptune",
+    diameter: "30,779 miles",
+    gravity: 1.21,
+    year_length: "16 hours, 7 mins",
+    distance_from_the_sun: "2797770000 miles",
+  },
+  {
+    name: "Mars",
+    diameter: "4,222 miles",
+    gravity: 0.38,
+    year_length: "24 hours, 37 mins",
+    distance_from_the_sun: "141700000 miles",
+  }
+]
+
+# Create a method to create a pool of Planet instance by using pool information which is an array
+def pool_to_planet(pool)
+  # Create a variable to store all new Planet s
+  library_of_planets = []
+  pool.each do |planet_info|
+    library_of_planets << planet = Planet.new(planet_info[:name], planet_info[:diameter], planet_info[:gravity],
+      planet_info[:year_length], planet_info[:distance_from_the_sun])
+    end
+    return library_of_planets
+  end
+
+  library = SolarSystem.new(pool_to_planet(pool))
+
+  # Create a user interface allowing user to interact with the solar system
+  # This program allow user to learn attributes of a planet and create their own planet
+  puts "Welcome to the Fun Solar System program!
+  Here not only you can learn planets from our library,
+  but also you can create your own planet."
+
+  flag = true
+  while flag
+    puts "\n********** I AM LIBRARY **********"
+    puts library.show_library
+    library.name_pool << "Create"
+    library.name_pool << "Exit"
```

Comment:
This use of the `name_pool` is interesting. It's a clever idea, but you end up mixing up user interface data with solar system data, and with a larger application this sort of thing can get very messy very fast. Better, I think, to let a `SolarSystem` keep track of planets and nothing else.

Reply:
I didn't quite understand this problem until I read your comments and the second chapter of OODR. I will always keep single responsibility principal in mind.